### ISSUE TITLE
Magnificent Arctic Guppy - Users can be able to withdraw more rewards than supposed

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-06-tokemak-judging#373]
Magnificent Arctic Guppy

high

# Users can be able to withdraw more rewards than supposed
## Summary
 No check for existing extraRewards during push
## Vulnerability Detail
POC:
1. Let us Assume MainRewarder had mistakenly added the same address twice in addExtraReward()
2. User calls stake() , linked rewards is staked twice to the same address
3. Now, user calls withdraw() to withdraw linked rewards (this is already 2x in step 2)
User will receive double the linked rewards due to the iteration in 
https://github.com/sherlock-audit/2023-06-tokemak/blob/main/v2-core-audit-2023-07-14/src/rewarders/MainRewarder.sol#L77C3-L79C10
## Impact
When adding extraRewards to the extra reward pool in https://github.com/sherlock-audit/2023-06-tokemak/blob/main/v2-core-audit-2023-07-14/src/rewarders/MainRewarder.sol#L58C1-L58C1 , there's no check for already existing address.
Assume a particular address takes up 2 slots out of 3, and a user withdraws staked extra rewards, the user will receive double the amount requested in https://github.com/sherlock-audit/2023-06-tokemak/blob/main/v2-core-audit-2023-07-14/src/rewarders/MainRewarder.sol#L77C3-L79C10
## Code Snippet
https://github.com/sherlock-audit/2023-06-tokemak/blob/main/v2-core-audit-2023-07-14/src/rewarders/MainRewarder.sol#L58C1-L58C1

https://github.com/sherlock-audit/2023-06-tokemak/blob/main/v2-core-audit-2023-07-14/src/rewarders/MainRewarder.sol#L77C3-L79C10
In summary this will ultimately lead to loss of funds
## Tool used
Manual Review + H2O 

## Recommendation
Add a check for an already existing extraRewards before line 58
